.gitlab-ci.yml

```diff
@@ -62,12 +62,18 @@ trigger-agent-build:
     - export SLACK_CACHE_DIR="${PWD}/.slack-cache"
     - DATADOG_AGENT_PIPELINE_URL=$(aws ssm get-parameter --region us-east-1 --name ci.integrations-core.datadog-agent-pipeline-url --with-decryption --query "Parameter.Value" --out text)
     - export AGENT_REQUIREMENTS_CHANGED="false"
+    - LATEST_TAG=$(git describe --tags --abbrev=0 HEAD)
     - git fetch origin master
-    - git --no-pager diff --exit-code --name-only HEAD $(git merge-base HEAD FETCH_HEAD) -- datadog_checks_base/datadog_checks/base/data/agent_requirements.in || export AGENT_REQUIREMENTS_CHANGED="true" || true
+    - BASE_REF=$(git merge-base HEAD FETCH_HEAD $LATEST_TAG)
```

Comment:
Get latest common commit between `HEAD` and master (`FETCH_HEAD`), and `HEAD` and `LATEST_TAG`.

Reply:
2 cases:
* We branched off from `LATEST_TAG=integration-1.2.3` ; in that case the python script is checking that it does start by `7.x` so it's ignoring the tag
* We branched off from commits after `LATEST_TAG` ; in that case we set `LATEST_TAG=main` just below

Reply:
But in those cases, will the pipeline still run off changes based on master?

Reply:
yes https://github.com/DataDog/integrations-core/pull/10842/files#diff-b4d35ba9894a41c951ead81b42de092804a0bff4c23d60660c00a6cbddd639e6R14-R16

Reply:
I feel like this approach may not be the best then unfortunately, there seem to be too many edge cases

Reply:
I would guess the release branch tag is the latest tag ~50% of the time when releasing, so for the other 50% of the time we will still run the pipeline on commits to the release branch

Reply:
Refactored the logic, closing this discussion